### PR TITLE
HDDS-10422. Fix some warnings about exposing internal representation in hdds-common

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Fix some Spotbugs (4.x) warnings that code `may expose internal representation`.  Notable changes:

- ~`OzoneQuota` used to initalize contents of `QuotaList`, moved to the latter~ (deferred to HDDS-10896)
- ~move creation of mutable metrics objects to `PerformanceMetrics`~ (deferred to HDDS-10897)

Also remove unused `hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/BatchOperation.java` (seems to be outdated duplicate of `hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/BatchOperation.java`).

https://issues.apache.org/jira/browse/HDDS-10422

## How was this patch tested?

CI:
https://github.com/adoroszlai/ozone/actions/runs/8200024023

### CODE REVIEW

**Review by szetszwo — COMMENTED** [review]
@adoroszlai , Thanks for working on this!  Please see the comments inlined and also https://issues.apache.org/jira/secure/attachment/13068960/6351_review.patch

BTW, this change is quite big.  How about separating it into at least two JIRAs?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/container/common/helpers/ChunkInfo.java [hunk 1]
```diff
@@ -97,10 +93,6 @@ public static ChunkInfo getFromProtoBuf(ContainerProtos.ChunkInfo info)
     chunkInfo.setChecksumData(
         ChecksumData.getFromProtoBuf(info.getChecksumData()));
 
-    if (info.hasStripeChecksum()) {
-      chunkInfo.setStripeChecksum(info.getStripeChecksum());
-    }
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/container/common/helpers/ChunkInfo.java:102]: No, `ChunkInfo` proto may have stripe checksum.  However, `getStripeChecksum` of the non-proto `ChunkInfo` is never called.  I'll remove this part of the change.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java [hunk 2]
```diff
@@ -222,7 +222,7 @@ public synchronized void setPort(Name name, int port) {
    * @return DataNode Ports
    */
   public synchronized List<Port> getPorts() {
-    return ports;
+    return new ArrayList<>(ports);
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java:245]: Thanks for the suggestion, updated.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java [hunk 3]
```diff
@@ -44,27 +39,17 @@ public enum Units { B, KB, MB, GB, TB }
...
-  /** Setting QuotaList parameters from large to small. */
-  static {
-    quotaList = new QuotaList();
-    quotaList.addQuotaList(OZONE_QUOTA_TB, Units.TB, TB);
-    quotaList.addQuotaList(OZONE_QUOTA_GB, Units.GB, GB);
-    quotaList.addQuotaList(OZONE_QUOTA_MB, Units.MB, MB);
-    quotaList.addQuotaList(OZONE_QUOTA_KB, Units.KB, KB);
-    quotaList.addQuotaList(OZONE_QUOTA_B, Units.B, 1L);
-  }
+  private static final QuotaList QUOTA_LIST = new QuotaList();
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java]: Deferred to HDDS-10897.

**Comment by adoroszlai** [comment]
Thanks @szetszwo for the review.  Deferred some changes that can be logically grouped to separate tasks: HDDS-10896, HDDS-10897 and HDDS-10899.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by adoroszlai** [comment]
Thanks @szetszwo for reviewing and merging this.

**Pull request merged**